pilot/pkg/config/kube/ior/route.go

```diff
@@ -104,10 +89,9 @@ func newRoute(
 	pilotNamespace string,
 	mrc controller.MemberRollController,
 	stop <-chan struct{},
-) *route {
-	for !kubeClient.IsRouteSupported() {
-		IORLog.Infof("routes are not supported in this cluster; waiting for Route resource to become available...")
-		time.Sleep(10 * time.Second)
+) (*route, error) {
+	if !kubeClient.IsRouteSupported() {
+		return nil, fmt.Errorf("routes are not supported in this cluster")
 	}
```

Comment:
Are you handling this elsewhere now? The purpose of the for loop was for the function to wait until the Route resource appears in the cluster. Previously, the function would return an error instead of waiting, but this was problematic during cold-starts of the cluster (see OSSM-1301).

Reply:
Make sure using `for` loop.